source/plot/svg/Shapes.ooc

```diff
@@ -1,31 +1,45 @@
-		This rect(upperRight x, upperRight y, size x, size y, opacity, color)
+	circle: static func (center: FloatPoint2D, r: Float, color: ColorBgra) -> String {
+		This ellipse(center, r, r, color)
 	}
-	rect: static func ~positionFloatPoint2D (upperRight: FloatPoint2D, width, height, opacity: Float, color: String) -> String {
-		This rect(upperRight x, upperRight y, width, height, opacity, color)
+	ellipse: static func (center: FloatPoint2D, rx, ry: Float, color: ColorBgra) -> String {
+		"<ellipse cx='" << center x toString() >> "' cy='" & center y toString() >> "' rx='" & rx toString() >> "' ry='" & ry toString() >> "' fill='" >> This getColor(color) >> "' fill-opacity='" >> This getOpacity(color) >> "'/>\n"
 	}
-	rect: static func ~Float (x, y, width, height, opacity: Float, color: String) -> String {
-		"<rect x='" << x toString() >> "' y='" & y toString() >> "' width='" & width toString() >> "' height='" & height toString() >> "' fill='" >> color >> "' fill-opacity='" & opacity toString() >> "'/>\n"
+	rect: static func ~FloatPoint2D (upperRight, size: FloatPoint2D, color: ColorBgra) -> String {
+		This rect(upperRight x, upperRight y, size x, size y, color)
 	}
-	text: static func ~simple (position: FloatPoint2D, text: String, fontSize: Int, opacity := 255.0f, color := "black") -> String {
-		This text(position, text, fontSize, "start", opacity, color)
+	rect: static func ~positionFloatPoint2D (upperRight: FloatPoint2D, width, height: Float, color: ColorBgra) -> String {
+		This rect(upperRight x, upperRight y, width, height, color)
 	}
-	text: static func ~textAnchor (position: FloatPoint2D, text: String, fontSize: Int, textAnchor: String, opacity := 255.0f, color := "black") -> String {
-		"<text id='" + text >> "' x='" & position x toString() >> "' y='" & position y toString() >> "' font-size='" & fontSize toString() >> "px' text-anchor='" >> textAnchor >> "' fill='" >> color >> "' fill-opacity='" & opacity toString() >> "'>" >> text >> "</text>\n"
+	rect: static func ~Float (x, y, width, height: Float, color: ColorBgra) -> String {
+		"<rect x='" << x toString() >> "' y='" & y toString() >> "' width='" & width toString() >> "' height='" & height toString() >> "' fill='" >> This getColor(color) >> "' fill-opacity='" >> This getOpacity(color) >> "'/>\n"
 	}
+	text: static func ~simple (position: FloatPoint2D, text: String, fontSize: Int, color := ColorBgra new(0, 0, 0, 255)) -> String {
+		This text(position, text, fontSize, "start", color)
+	}
+	text: static func ~textAnchor (position: FloatPoint2D, text: String, fontSize: Int, textAnchor: String, color := ColorBgra new(0, 0, 0, 255)) -> String {
+		"<text id='" + text >> "' x='" & position x toString() >> "' y='" & position y toString() >> "' font-size='" & fontSize toString() >> "px' text-anchor='" >> textAnchor >> "' fill='" >> This getColor(color) >> "' fill-opacity='" >> This getOpacity(color) >> "'>" >> text >> "</text>\n"
+	}
+}
+
+Shape: enum {
```

Comment:
Could it be possible to use a base class and subclasses for every shape instead of `enum` like this ? Just asking because @davidhesselbom likes object-oriented style :)